api/v3/utils.php

```diff
@@ -176,8 +176,13 @@ function civicrm_api3_create_success($values = 1, $params = array(), $entity = N
         $values[$key]['id'] = $item[$lowercase_entity . "_id"];
       }
       if (!empty($item['financial_type_id'])) {
-        //4.3 legacy handling
-        $values[$key]['contribution_type_id'] = $item['financial_type_id'];
+        // 4.3 legacy handling - translate financial_type to contribution_type unless financial_type is explicitly specified.
```

Comment:
my theory is that it is expecting that contribution_type_id should be outputted in the api array and always matches that of the financial_type_id